src/test/mocks/archiveLoader.ts

```diff
@@ -85,13 +85,25 @@ export default () => {
       localBooks[`${newBook.id}@${newBook.version}`] = newBook;
       localBookPages[`${newBook.id}@${newBook.version}`] = {};
     },
-    mockPage: (parentBook: ArchiveBook, newPage: ArchivePage, pageSlug: string) => {
-      localBookPages[`${parentBook.id}@${parentBook.version}`][newPage.id] = newPage;
-      localBooks[`${parentBook.id}@${parentBook.version}`].tree.contents.push({
+    // tslint:disable-next-line: max-line-length
+    mockPage: (parentBook: ArchiveBook, newPage: ArchivePage, pageSlug: string, parentChapter?: ArchiveTree, parentSection?: ArchiveTree) => {
```

Comment:
@TomWoodward curious if there is a better way to do this or if I should write a new function. `mockPage()` was built to add a page at the book level and I wanted options for adding book subtrees.

Reply:
it looks like perhaps you don't even need to call `mockPage` because you're setting up your tree structure manually right?

Reply:
I'm trying to implement it this way but without `mockPage` the loader just has empty pages via `mockBook`: 
```
mockBook: (newBook: ArchiveBook) => {
      localBooks[`${newBook.id}@${newBook.version}`] = newBook;
      localBookPages[`${newBook.id}@${newBook.version}`] = {};
    },
```

This causes `getCleanContent()` to return empty because each page cache returns undefined: 
```
const cachedPage = book && page &&
    archiveLoader.book(book.id, book.version).page(page.id).cached()
  ;
```

Reply:
oh yea, `mockPage` does some stuff to the tree that you've already handled and is probably bad for it to try to do again, but you need the part where it updates `localBookPages`.

you may need to update `mockPage` to conditionally not update the tree